### PR TITLE
Feature:UserTimeZone

### PR DESCRIPTION
**Pull request by samhithaarani**
Genaral Approch:
listens for user input in the search input field.
As the user types, it filters the options in the dropdown based on whether they contain the search term.
If an option matches the search term, it's displayed; otherwise, it's hidden.

solved #10186 
resolved

### CODE REVIEW

**Comment by michaelkrar** [comment]
@samhithaarani Reviewing this right now everything looks good but I do not understand how the commits in DatabaseQueries.php help to solve the issue or if it is a solution to another issue, as this looks like code that deals with Gradeable data rather than User data(correct me if I am wrong).

**Review by michaelkrar — CHANGES_REQUESTED** [review]
See above comment

**Comment by samhithaarani** [comment]
> @samhithaarani Reviewing this right now everything looks good but I do not understand how the commits in DatabaseQueries.php help to solve the issue or if it is a solution to another issue, as this looks like code that deals with Gradeable data rather than User data(correct me if I am wrong). When I undo the changes done to that file the fix still works properly too.

Hey :) Thanks for your review....ther's nothing to do with the DatabaseQueries.php , i've got this unknown Deprecated error,so i've done this modification to ensures that json_decode() is not called with null, thus avoiding the deprecated behavior and addressing the issue ...... Thank you :)
Issue-Submitty

**Comment by IDzyre** [comment]
> >
> Hey :) Thanks for your review....ther's nothing to do with the DatabaseQueries.php , i've got this unknown Deprecated error,so i've done this modification to ensures that json_decode() is not called with null, thus avoiding the deprecated behavior and addressing the issue ...... Thank you :)

This should be in a different PR in that case, since it is not related to the time zone issue you are trying to resolve.

**Comment by samhithaarani** [comment]
> See above comment. also change title to include "[Feature:..."

Hey i'm sorry not aware of it ...i'm still learning! 
can i create a new issue and implement separate PR's  ?

File: site/app/libraries/database/DatabaseQueries.php [hunk 1]
```diff
@@ -6207,10 +6207,14 @@ function ($value) {
             );
 
             // Unpack the component data
+            // updated the deprecated behavior in PHP. Passing null to json_decode()
```
Comment (04vmatsibekker) [inline site/app/libraries/database/DatabaseQueries.php:6210]: remove this line

File: site/app/libraries/database/DatabaseQueries.php [hunk 2]
```diff
@@ -6207,10 +6207,14 @@ function ($value) {
...
             // Unpack the component data
+            // updated the deprecated behavior in PHP. Passing null to json_decode() 
             $unpacked_component_data = [];
             foreach (array_merge($component_properties, $component_mark_properties) as $property) {
-                $unpacked_component_data[$property] = json_decode($row['array_' . $property]) ?? [];
+                $json_data = $row['array_' . $property];
+                $decoded_data = ($json_data !== null) ? json_decode($json_data) : [];
+                $unpacked_component_data[$property] = $decoded_data;
             }
+
```
Comment (04vmatsibekker) [inline site/app/libraries/database/DatabaseQueries.php:6217]: remove this line

File: site/app/templates/UserProfile.twig [hunk 3]
```diff
@@ -140,10 +140,15 @@
...
-                    </div>
+                    <label for="time_zone_drop_down">Select Timezone: </label>
+                    <input type="text" id="time_zone_search" placeholder="Search Timezone">
+                    <select id="time_zone_drop_down" data-testid="time-zone-dropdown">
+                        {% for timezone in available_time_zones %}
+                            <option>{{ timezone }}</option>
+                        {% endfor %}
+                    </select>
                 </div>
+
```
Comment (04vmatsibekker) [inline site/app/templates/UserProfile.twig:151]: remove extra line

File: site/app/templates/UserProfile.twig [hunk 4]
```diff
@@ -218,6 +223,40 @@
         </div>
     </div>
 </div>
+<script>
```
Comment (04vmatsibekker) [inline site/app/templates/UserProfile.twig:226]: I think you should move the code within the script tag into user-profile.js .

**Comment by 04vmatsibekker** [comment]
Also the title was done incorrectly. Here is the submitty documentation for properly naming a PR: https://submitty.org/developer/getting_started/make_a_pull_request

**Comment by IDzyre** [comment]
was this added in https://github.com/Submitty/Submitty/pull/10277

**Comment by dorian451** [comment]
Yeah, it seems we already have this feature in https://github.com/Submitty/Submitty/pull/10277. Thanks for your work!

**Pull request closed without merging**